rfc/attestation-config.md

```diff
@@ -0,0 +1,76 @@
+# Attestation configuration options
+
+To allow users more in-depth control over validating attestation statements, a separate, attestation-specific entry in the user's Constellation config file needs to be added.
+
+This option should allow for more in-depth configuration of how and what parts of an attestation statement are validated.
+For example, one should be able to set the minimum microcode version of an SEV-SNP attestation statement, that is deemed acceptable,
+or configure acceptable TCB statuses for Intel TDX attestation.
+
+## In the user's config
+
+A new `attestation` entry in the config file will be added, holding the options used for attestation validation.
+Existing attestation options, like `measurements` and `idKeyDigests` will be moved to this entry.
+
+This entry will replace the `attestationVariant` entry.
+
+Example configuration for Azure SEV-SNP:
+
+```yaml
+version: v3
+image: vX.Y.Z
+name: constell
+stateDiskSizeGB: 30
+kubernetesVersion: "v1.25.6"
+debugCluster: false
+provider:
+  azure: {} # Configuration for creating Azure cloud resources
+# Configuration for attestation validation. This configuration provides sensible defaults for the Constellation version it was created for.
+# See our docs for an overview on attestation: https://docs.edgeless.systems/constellation/architecture/attestation
+attestation:
+  # Azure SEV-SNP attestation configuration
+  # See our docs for information on about configurable values [Link to our docs explaining SEV-SNP on Azure]
+  azureSEVSNP:
+    # Expected confidential VM image measurements.
+    measurements:
+      15:
+        expected: "0000000000000000000000000000000000000000000000000000000000000000"
+        warnOnly: false
+    # Configuration for validating the firmware signature.
+    firmwareSignerConfig:
+      # List of accepted values for the firmware signing key digest.
+      acceptedKeyDigests:
+        - 7486a447ec0f1958002a22a06b7673b9fd27d11e1c6527498056054c5fa92d23c50f9de44072760fe2b6fb89740cc96
+      # Key digest enforcement policy. One of {'equal', 'maaFallback', 'warnOnly'}
+      #     - 'equal'       : Error if the reported signing key digest does not match any of the values in 'acceptedKeyDigests'
+      #     - 'maaFallback' : Use 'equal' checking for validation, but fallback to using MAA for validation if the reported digest does not match any of the values in 'acceptedKeyDigests'
+      #     - 'warnOnly'    : Same as 'equal', but only prints a warning instead of returning an error if no match is found
+      enforcementPolicy: maaFallback
+    # Lowest acceptable microcode version
+    microcodeVersion: 115
+    # Lowest acceptable SNP version
+    snpVersion: 8
+    # Lowest acceptable bootloader version
+    bootloaderVersion: 3
```

Comment:
For these, we need some form of auto update for users that don't touch attestation config. Currently, they get updates for these values by updating to a new Constellation version.

Reply:
We could implement this as part of a config migration step when using a new cli version for the first time.

Reply:
Could also be implemented using the versions API couple with a CLI command

Reply:
Another idea: omit non-required values when creating a config with `generate`. For those omitted values, use the the CLI's defaults. The user learns about the config values through the docs. Wouldn't need the migration step and users who don't care automatically use the latest version for each CLI.

But Malte's and Daniel's ideas are closer to what we are currently doing.

Reply:
> Another idea: omit non-required values when creating a config with generate. For those omitted values, use the the CLI's defaults. The user learns about the config values through the docs. Wouldn't need the migration step and users who don't care automatically use the latest version for each CLI.

I like the idea 🤩. What about a sentinel value `auto` or `managed` (or similar) to indicate that the CLI will take care of upgrading the value?

Reply:
Added specification for how I imagine this API to look like